TitanSchoolsClient.js

```diff
@@ -80,20 +83,22 @@ class TitanSchoolsClient {
    *   }
    * ]
    */
-  async fetchMenu(startDate = null) {
+  async fetchMenu(startDate = null, endDate = null) {
     let params = {
       ...this.requestParams,
       // If no startDate was provided, use today's date
       // API requires date to be formatted as: m-d-Y (i.e. 12-5-2021)
-      startDate: this.formatDate(startDate ?? new Date(Date.now())),
+      startDate: this.formatDate(startDate ?? moment()),
+      endDate: this.formatDate(endDate ?? moment().add(7, "days"))
```

Comment:
I think there's a bug hiding in here:

- When the `endDate` is provided as an argument, this logic will actually add 7 days onto the requested endDate instead of using the date provided.
- When the `endDate` isn't provided and we need to come up with a value, we _always_ add 7 days instead of using the `config.numberOfDaysToDisplay` value.

In the first case, we simply overfetch data from the API and then filter out the extra days in the `processData()` method, so no real harm done.
But in the case where we set the endDate to be 7 days from today, we won't have enough data from the API if the module is configured with `config.numberOfDaysToDisplay` greater than 7 (e.g. the user wants 8 days to be visible).

I looked at the changes that this PR makes to the `node_helper.js` and I think all of the logic for determining the `endDate` actually looks correct to me in there.  Since those values for `startDate` and `endDate` are the ones that get passed into this `client.fetchMenu()` method, I think it would actually make sense to just convert `startDate` and `endDate` to be required arguments and then just eliminate all of the logic that tries to derive the correct values from within this `TitanSchoolsClient.fetchMenu()` method altogether.

Let me know what you think about that proposal - my hope is that it will fix this current bug while also reducing the number of places where we could potentially introduce a bug in the future.

Reply:
I meant to update this to use `config.numberOfDaysToDisplay`, however, I think requiring these parameters is a better solution.